packages/pwa-kit-react-sdk/src/ssr/server/opentelemetry-server.js

```diff
@@ -0,0 +1,172 @@
+ * @param {string} [options.serviceVersion]
+ * @param {boolean} [options.enabled]
+ * @returns {NodeTracerProvider|null}
+ */
+export const initializeServerTracing = (options = {}) => {
+    const {
+        serviceName = options.serviceName || getServiceName(),
+        serviceVersion,
+        enabled = getOTELConfig().enabled
+    } = options
+    // If tracing is disabled, return null without initializing
+    if (!enabled) {
+        return null
+    }
+
+    try {
+        // Build resource attributes
+        const resourceAttributes = {
+            'service.name': serviceName
+        }
+
+        // Add service version if provided
+        if (serviceVersion) {
+            resourceAttributes['service.version'] = serviceVersion
+        }
+
+        // Initialize the tracer provider
+        provider = new NodeTracerProvider({
+            resource: new Resource(resourceAttributes),
+            spanProcessor: new SimpleSpanProcessor()
+        })
+
+        // Add B3 propagator
+        propagation.setGlobalPropagator(new B3Propagator())
+        provider.register()
+
+        return provider
+    } catch (error) {
+        // Log errors from OpenTelemetry initialization
+        logger.error('Failed to initialize OpenTelemetry provider', {
+            namespace: 'opentelemetry-server.initializeServerTracing',
+            additionalProperties: {error: error.message}
+        })
+        return null
+    }
+}
+
+/**
+ * Shutdown OpenTelemetry tracing and clean up resources
+ * @returns {Promise<void>}
+ */
+export const shutdownServerTracing = async () => {
+    if (provider) {
+        try {
+            await provider.shutdown()
+            provider = null // Clean up after successful shutdown
+        } catch (error) {
+            logger.warn('Failed to shutdown OpenTelemetry provider', {
+                namespace: 'opentelemetry-server.shutdownServerTracing',
+                additionalProperties: {error: error.message}
+            })
+        }
+    }
+}
+
+/**
+ * Get the current OpenTelemetry provider instance
+ * @returns {NodeTracerProvider|null} The current provider or null if not initialized
+ */
+export const getServerTracingProvider = () => {
+    return provider
+}
+
+/**
+ * Check if OpenTelemetry tracing is currently initialized
+ * @returns {boolean} True if tracing is initialized, false otherwise
+ */
+export const isServerTracingInitialized = () => {
+    return provider !== null
+}
+
+/**
+ * Creates a span for performance measurement
+ * @param {string} name - The name of the performance span
+ * @param {Function} fn - The function to measure
+ * @param {Object} res - The response object (optional)
+ * @returns {Promise<any>} The result of the function
+ */
+export const tracePerformance = async (name, fn, res = null, req = null) => {
+    // Check if OpenTelemetry is properly configured
+    const otelConfig = getOTELConfig()
+    let rootSpan = null
+    let ctx = null
+    if (otelConfig.enabled) {
```

Comment:
nit: We can simplify if move the `shouldTrackPerformance` to this conditional to avoid creating tracing objects when tracing is not enabled.
```suggestion
    if (otelConfig.enabled && shouldTrackPerformance) {
```